restapi/src/main/java/io/stargate/web/docsapi/exception/ErrorCode.java

```diff
@@ -101,9 +101,9 @@
   DOCS_API_SEARCH_OBJECT_REQUIRED(
       Response.Status.BAD_REQUEST, "Search was expecting a JSON object as input."),
 
-  DOCS_API_SEARCH_OR_NOT_SUPPORTED(
-      Response.Status.BAD_REQUEST,
-      "Searching documents with the $or condition is not yet supported."),
+  DOCS_API_SEARCH_EXPRESSION_NOT_RESOLVED(
+      Response.Status.INTERNAL_SERVER_ERROR,
+      "Can not find the document resolver for the given expression."),
```

Comment:
IMHO, the new message is a bit cryptic for the end-user. How about something like `Unable to resolve the given expression`?

Also, is it really a server error? I'd say if we're not able to process something because of a feature that is know to be not implemented, and we can detect that condition, we ought to report `BAD_REQUEST`. WDYT?

Reply:
This is really a server exception and should be reported like this so that we can look into it.. Should never happen, but I need to throw something here.. I'll update the message.. :+1: